modules/AlphaWalletFoundation/AlphaWalletFoundation/Extensions/Publishers/Session+Publishers.swift

```diff
@@ -12,13 +12,26 @@ import Combine
 
 extension APIKitSession {
 
-    class func sendPublisher<Request: APIKit.Request>(_ request: Request, server: RPCServer, callbackQueue: CallbackQueue? = nil) -> AnyPublisher<Request.Response, SessionTaskError> {
+    class func sendPublisher<Request: APIKit.Request>(_ request: Request, server: RPCServer, analytics: AnalyticsLogger, callbackQueue: CallbackQueue? = nil) -> AnyPublisher<Request.Response, SessionTaskError> {
         sendImplPublisher(request, server: server, callbackQueue: callbackQueue)
             .retry(times: 2, when: {
                 guard case SessionTaskError.requestError(let e) = $0 else { return false }
+                if let e = e as? RpcNodeRetryableRequestError {
+                    logRpcNodeError(e, analytics: analytics)
+                }
                 return e is RpcNodeRetryableRequestError
-            })
-            .eraseToAnyPublisher()
+            }).eraseToAnyPublisher()
+    }
+
+    private static func logRpcNodeError(_ rpcNodeError: RpcNodeRetryableRequestError, analytics: AnalyticsLogger) {
```

Comment:
Is the implementation of `logRpcNodeError(_:,analytics:) `the same as the one in `Session+PromiseKit.swift`? Then we can call it instead?